packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+      suspendedByProducer: false,
+      suspendedByPlatform: false,
+    };
+
+    const attribute: Attribute = {
+      id: validVerifiedTenantAttribute.id,
+      kind: attributeKind.verified,
+      description: "A verified attribute",
+      name: "A verified attribute name",
+      creationTime: new Date(new Date().getFullYear() - 1),
+    };
+
+    await addOneEService(eservice);
+    await addOneTenant(consumer);
+    await addOneTenant(producer);
+    await addOneAttribute(attribute);
+    await addOneAgreement(agreement);
+
+    const {
+      archivableRelatedAgreement1,
+      archivableRelatedAgreement2,
+      nonArchivableRelatedAgreement,
+    } = await addRelatedAgreements(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+    const mockUserResponse: UserResponse = {
+      email: "selfcare.test.submitagreement@test.org",
+      name: "Test Name",
+      surname: "Test Surname",
+      id: generateId(),
+      taxCode: "TAXCODE",
+    };
+
+    mockSelfCareResponse(mockUserResponse);
+    const submittedAgreement = await agreementService.submitAgreement(
+      agreement.id,
+      {
+        consumerNotes: consumerNotesText,
+      },
+      {
+        authData,
+        correlationId: randomUUID(),
+        serviceName: "AgreementServiceTest",
+        logger: genericLogger,
+      }
+    );
+
+    expect(submittedAgreement).toBeDefined();
+    expect(submittedAgreement.state).toBe(agreementState.active);
+
+    const uploadedFiles = await fileManager.listFiles(
+      config.s3Bucket,
+      genericLogger
+    );
+
+    expect(submittedAgreement.contract).toBeDefined();
+    expect(uploadedFiles.length).toEqual(0);
+
+    const actualAgreementData = await readLastAgreementEvent(agreement.id);
+    if (!actualAgreementData) {
+      fail("Creation fails: agreement not found in event-store");
+    }
+
+    expect(actualAgreementData.type).toEqual("AgreementActivated");
+    expect(actualAgreementData).toMatchObject({
+      type: "AgreementActivated",
+      event_version: 2,
+      version: "1",
+      stream_id: submittedAgreement.id,
+    });
+
+    const actualAgreement: AgreementV2 | undefined = decodeProtobufPayload({
+      messageType: AgreementActivatedV2,
+      payload: actualAgreementData.data,
+    }).agreement;
+
+    if (!actualAgreement) {
+      fail("impossible to decode AgreementAddedV1 data");
+    }
+
+    expect(selfcareV2ClientMock.getUserInfoUsingGET).not.toHaveBeenCalled();
+    expect(actualAgreement).toMatchObject({
+      id: submittedAgreement.id,
+      eserviceId: eservice.id,
+      descriptorId: descriptor.id,
+      producerId: producer.id,
+      consumerId: consumer.id,
+      state: toAgreementStateV2(agreementState.active),
+      contract: {
+        id: expect.any(String),
+        name: expect.any(String),
+        prettyName: expect.any(String),
+        contentType: expect.any(String),
+        path: expect.any(String),
+        createdAt: expect.any(BigInt),
```

Comment:
Here you could check the actual values instead of `any`, to test that the contract creation worked as expected (i.e.,  created a contract in the right path, etc.). I did a similar thing in #567

Reply:
In this case the submitted agreement has related agreement and doesn't create a new contract but using the same data, so I'm not interested to check mock date created before.